server/views/pages/visit/summary.njk

```diff
@@ -20,6 +22,39 @@
 
       <h1 class="govuk-heading-l">{{ pageHeaderTitle }}</h1>
 
+       {% if visit.visitStatus === "CANCELLED" %}
+        {% set status = visit.outcomeStatus.split("_") %}
+            {%- for visitNote in visit.visitNotes %}
+              {% if visitNote.type == "VISIT_OUTCOMES" %}
+
+                {% set cancelledBy %}
+                  {% if status[0] === 'ADMINISTRATIVE' %}
+                    {{ 'because of an administrative error.' }}
+                  {% else %}
+                    {{ 'by the ' + status[0] | lower  + '.' }}
+                  {% endif %}
+                {% endset %}
+
+                {% set cancelledVisitHtml %}
+                  {{ govukWarningText({
+                    classes: "app-warning-text--long govuk-!-padding-left-4",
+                    html: "<span>This visit was cancelled " + cancelledBy +"</span><br><span>Reason: " + visitNote.text + ".</span>",
```

Comment:
I would remove the `.` at the end where you have `".</span>"` - otherwise you end up with two if the user entered a full stop themselves!

Reply:
hadn't thought of that!